src/Model/GroundWaterFlow/gwf3sto8.f90

```diff
@@ -339,15 +390,15 @@ subroutine sto_fc(this, kiter, hold, hnew, njasln, amat, idxglo, rhs)
           if (this%iconf_ss == 0) then
             zold = bt + DHALF * tthk * snold
             znew = bt + DHALF * tthk * snnew
-            rhsterm = -rho1 * (snold * (hold(n) - zold) + snnew * znew)
+            rhsterm = -rho1old * snold * (hold(n) - zold) - rho1 * snnew * znew
           else
-            rhsterm = -rho1 * ssh0 + rho1 * ssh1
+            rhsterm = -rho1old * ssh0 - rho1 * ssh1
```

Comment:
If the above is not correct the sign of the second term need to be the same as before.